IGNITE-16045 Add documentation pages for the snapshot thread pool size property (#9624)
apache/ignite: docs/_docs/snapshots/snapshots.adoc

The link:persistence/snapshots#restoring-from-snapshot[restore procedure] explains how to tether together all the segments during recovery.
====

== Configuration

=== Snapshot Directory

By default, a segment of the snapshot is stored in the work directory of a respective Ignite node and uses the same storage
media where Ignite Persistence keeps data, index, WAL, and other files. Since the snapshot can consume as much space as
already taken by the persistence files and can affect your application's performance by sharing the disk I/O with the
Ignite Persistence routines, it's suggested to store the snapshot and persistence files on different media.

See the link:persistence/snapshot-directory#configuring-snapshot-directory[Configuring Snapshot Directory] page for
configuration examples.

=== Snapshot Execution Pool

> By default, the snapshot thread pool size has value `4`. Decreasing the number of threads involving into snapshot creation will
> increase the total amount of time the snapshot being taking, however, this will keep the disk load within reasonable limits.

Reviewer: ```suggestion
By default, the snapshot thread pool size has a value of `4`. Decreasing the number of threads involved in the snapshot creation process 
increases the total amount of time for taking a snapshot. However, this keeps the disk load within reasonable limits.
```

> See the link:perf-and-troubleshooting/thread-pools-tuning[Ignite Snapshot Execution Pool] page.

Reviewer: ```suggestion
See the link:perf-and-troubleshooting/thread-pools-tuning[Ignite Snapshot Execution Pool,window=_blank] page for more details.
```
